Examples: adding basic datasource example (#16)
grafana/grafana-plugin-examples: README.md

- [http](examples/http-datasource) demonstrates how to query data from HTTP-based APIs.
- [alerting](examples/alerting) demonstrates how to add support for alerting to a data source, using a backend.
- [streaming-websocket](examples/streaming-websocket) demonstrates how to create an event-based data source plugin using RxJS and web sockets.

> - [basic-datasource](examples/basic-datasource) demonstrates how to build a datasource that supports the most basic functionality of Grafana.

Reviewer: ```suggestion
- [basic-datasource](examples/basic-datasource) demonstrates how to build a basic data source plugin.
```

Reviewer: Maybe this should be moved to first in the list as it's likely the simplest? Was thinking it could be nice to mention it demonstrates 2 different query paths and also health checking, but not sure how to best word it simply.

Maybe the HTTP example could also be
```
- [http](examples/http-datasource)  demonstrates how to build a data source plugin that queries data from HTTP-based APIs.
```

Author: Yes, agree!
